server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequestParameters.java

```diff
@@ -92,22 +96,50 @@ public enum Metric {
         REPOSITORIES("repositories"),
         ALLOCATIONS("allocations");
 
-        private String metricName;
+        public static final Set<Metric> ALL = Collections.unmodifiableSet(EnumSet.allOf(Metric.class));
+        public static final Set<String> ALL_NAMES = ALL.stream().map(Metric::metricName).collect(toUnmodifiableSet());
+        public static final Map<String, Metric> NAMES_MAP = ALL.stream().collect(toUnmodifiableMap(Metric::metricName, m -> m));
+        private final String metricName;
 
-        Metric(String name) {
-            this.metricName = name;
+        Metric(String metricName) {
+            this.metricName = metricName;
         }
 
-        public String metricName() {
-            return this.metricName;
+        public static boolean isValid(String name) {
+            return NAMES_MAP.containsKey(name);
+        }
+
+        public static Metric get(String name) {
+            return NAMES_MAP.get(name);
         }
 
-        boolean containedIn(Set<String> metricNames) {
-            return metricNames.contains(this.metricName());
+        public static void writeSetTo(StreamOutput out, EnumSet<Metric> metrics) throws IOException {
+            if (out.getTransportVersion().onOrAfter(TransportVersions.USE_NODES_STATS_REQUEST_METRIC_ENUM)) {
+                out.writeEnumSet(metrics);
```

Comment:
Please do not rely on writeEnumSet/readEnumSet as they are using `Enum:ordinal()` that is easy to break simply my reordering enum values, see https://github.com/elastic/elasticsearch/pull/107944